pkg/logs/launchers/integration/launcher.go

```diff
@@ -47,22 +61,40 @@ func NewLauncher(sources *sources.LogSources, integrationsLogsComp integrations.
 	err := os.MkdirAll(runPath, 0755)
 
 	if err != nil {
-		ddLog.Warn("Unable to create integrations logs directory:", err)
+		ddLog.Error("Unable to create integrations logs directory:", err)
+	}
+
+	logsTotalUsageSetting := pkgconfigsetup.Datadog().GetInt64("logs_config.integrations_logs_total_usage") * 1024 * 1024
+	logsUsageRatio := pkgconfigsetup.Datadog().GetFloat64("logs_config.integrations_logs_disk_ratio")
+	maxDiskUsage, err := computeMaxDiskUsage(runPath, logsTotalUsageSetting, logsUsageRatio)
+	if err != nil {
+		ddLog.Warn("Unable to compute integrations logs max disk usage, defaulting to set value:", err)
+		maxDiskUsage = logsTotalUsageSetting
 	}
 
 	return &Launcher{
 		sources:              sources,
 		runPath:              runPath,
+		fileSizeMax:          pkgconfigsetup.Datadog().GetInt64("logs_config.integrations_logs_files_max_size") * 1024 * 1024,
+		combinedUsageMax:     maxDiskUsage,
+		combinedUsageSize:    0,
 		stop:                 make(chan struct{}),
 		integrationsLogsChan: integrationsLogsComp.Subscribe(),
 		addedConfigs:         integrationsLogsComp.SubscribeIntegration(),
-		integrationToFile:    make(map[string]string),
-		writeFunction:        writeLogToFile,
+		integrationToFile:    make(map[string]*FileInfo),
+		// Set the initial least recently modified time to the largest possible
+		// value, used for the first comparison
+		writeLogToFileFunction: writeLogToFile,
 	}
 }
 
 // Start starts the launcher and launches the run loop in a go function
 func (s *Launcher) Start(_ launchers.SourceProvider, _ pipeline.Provider, _ auditor.Registry, _ *tailers.TailerTracker) {
+	err := s.scanInitialFiles(s.runPath)
+	if err != nil {
+		ddLog.Warn("Unable to scan existing log files:", err)
+	}
```

Comment:
It could go either way, but I think it makes sense to continue even `scanInitialFiles` returns an error since you would still want logs from live integrations even if there's an issue scanning the initial files.